src/invisible/troubleshooting.md

```diff
@@ -21,15 +21,44 @@ Potential problems include:
 - Low storage space remaining on the Companion Device.
 - Unstable connection to the Pupil Invisible Glasses device during a recording.
 
-#### I accidentally updated my OnePlus 6 to an incompatible Android version!
+#### I accidentally updated my Companion device to an incompatible Android version!
 You need to rollback the Android version to be compatible again. Please note that all data on the phone will be lost during the rollback, so make backups accordingly.
 
-To execute the rollback, follow these steps:
+**OnePus 6**
+
+If you upgraded to Android 10 or newer and want to roll back, follow these steps:
 1. Download the ROM package [here](https://drive.google.com/file/d/1X8SVUH7UU7g0dcf-iWWXTPtMWzGguhUA/view?usp=sharing).
 1. Copy the ROM package to the mobile phone storage root directory.
 1. Go to Settings -> system updates -> top right corner icon -> local upgrade -> click on the installation package -> immediately upgrade -> system upgrade to 100%.
 1. Select Reboot system now.
- 
+
+**OnePlus 8/8T**
+
+If you upgraded to Android 12 or newer and want to roll back, follow these steps:
+1. Download the ROM package for your corresponding phone model to your computer
+
+   - [OnePlus 8](https://drive.google.com/file/d/1WtZ7bVwWPwYIGq4aWadeG-7MLu-KXhIS/view?usp=sharing)
+   - [OnePlus 8T](https://drive.google.com/file/d/1K2VI-R4gGN8mdS6FxLiRo48ICF_UBB7A/view?usp=sharing)
+
+2. Download the [official Rollback APK](https://oxygenos.oneplus.net/OPLocalUpdate_For_Android12.apk) to your computer
+3. Follow steps 1-5 from the [Transfer Exported Recordings to a Computer](/invisible/how-tos/data-collection-with-the-companion-app/transfer-recordings-via-usb.html#transfer-exported-recordings-to-a-computer) instructions
```

Comment:
Why do I need to do this? If it is only to backup recordings before wiping the phone this should not be mentioned here but above where backing up recordings is already mentioned IMO. This process is the same for OnePlus 6 and 8 too.

Other than that this is looking good I think!

Reply:
See my comment below